shipitscript/shipitscript/pushlog_scan.py

```diff
@@ -0,0 +1,98 @@
+from enum import Enum, auto, unique
+
+import requests
+
+URL = 'https://hg.mozilla.org/{repo}/json-pushes'
+
+
+@unique
+class Importance(Enum):
+    MAYBE = auto()
+    IMPORTANT = auto()
+    UNIMPORTANT = auto()
+
+
+_push_checks = []
+
+
+def push_check(func):
+    _push_checks.append(func)
+    return func
+
+
+def should_build(repo, old_rev, curr_rev):
+    # TODO redo
+    resp = requests.get(
```

Comment:
should we be using aiohttp if this is to be used by shipitscript and scriptworker? Both of which are async right?

Reply:
shipitscript uses requests in many places, decided to stick with it here (for now) -- but yes we should refactor to use async functions for this stuff.

Reply:
to be addressed in a follow-up. I'll open some issues for this repo once this is over.